Define separate `oracle::MonitorAttestations` for start-up

luckysori opened the pull request:
It should be more efficient to send one message instructing the actor to monitor several attestations when initialising the `oracle::Actor`, rather than multiple smaller messages.

Inspired by looking at #2049.

luckysori committed:
Extract body of oracle handler into Actor method

bonomat commented on daemon/src/oracle.rs:
```diff
@@ -38,6 +38,11 @@ pub struct MonitorAttestation {
     pub event_id: BitMexPriceEventId,
```
Do you think it makes sense to get rid of `MonitorAttestation` in the favor of `MonitorAttestations` and just send around a vec with 1 element?

luckysori replied:
I don't think it's a good idea for two reasons:

- The public API of this module doesn't need to let us send a message to monitor more than one attestation at a time.
- We would be allocating unnecessarily when sending `MonitorAttestations` with one item.

I think keeping both messages is worth it.

bonomat reviewed (approved):
Nice improvement.

bonomat commented on daemon/src/oracle.rs:
```diff
@@ -362,11 +362,13 @@ impl xtra::Actor for Actor {
+        self.tasks.add(async {
+            this.send_interval(SYNC_ATTESTATIONS_INTERVAL, || SyncAttestations)
+                .await;
+        })
```
This was on purpose in the same task to only start it after we fetched everything from the db.

luckysori replied:
Ooops, my bad. I thought there was no point and it made rebasing easier. I'll undo it!

bonomat reviewed (approved):
One comment. I'm not sure if it is a big difference if you keep it where it was.

luckysori committed:
Define separate oracle::MonitorAttestations message for start-up

It should be more efficient to send one message instructing the actor
to monitor several attestations when initialising the `oracle::Actor`,
rather than multiple smaller messages.
